src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -6094,5 +6094,57 @@ public class C
 
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
+
+        [Fact]
+        public void TestERR_ManagedAddr()
+        {
+            var comp = CreateCompilation(@"
+public unsafe struct S1
+{
+    public S1* s;
+    public object P { get => field; }
+}
+public unsafe struct S2
+{
+    public S2* s;
+    public int P { get => field; }
+}
+", options: TestOptions.UnsafeDebugDll);
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+
+            comp.VerifyDiagnostics(
+                // (4,16): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('S1')
+                //     public S1* s;
+                Diagnostic(ErrorCode.ERR_ManagedAddr, "s").WithArguments("S1").WithLocation(4, 16));
+
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestERR_ManagedAddr2()
+        {
+            var comp = CreateCompilation(@"
+public unsafe struct S1
+{
+    public object s;
+    public S1* P { get => field; }
+}
+public unsafe struct S2
+{
+    public int s;
+    public S2* P { get => field; }
+}
+", options: TestOptions.UnsafeDebugDll);
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+
+            comp.VerifyDiagnostics(
+                // (5,16): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('S1')
+                //     public S1* P { get => field; }
+                Diagnostic(ErrorCode.ERR_ManagedAddr, "P").WithArguments("S1").WithLocation(5, 16));
+
+            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
```

Comment:
>Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding); [](http://example.com/codeflow?start=12&length=70)

> There is no longer extra binding here. It's fixed in last few commits, but not sure the approach is good. 

It looks like the extra binding is still here.

Reply:
I think it's because I'm delaying `CheckAllConstraints` call for pointer types in fields only.

I'll need to do special case too many other symbols using one of these approaches:

- current approach (meaning updating the code in symbols and also special case them in MethodCompiler)
- updating code in symbols but get back to the virtual AfterAccessorBindingChecks that you requested to revert.

Do you like any of these approaches? Is there a completely different approach you'd like me to follow?

Reply:
> updating code in symbols but get back to the virtual AfterAccessorBindingChecks that you requested to revert. The reason it was reverted is we cared only about limited types of symbols, which will no longer be the case now.

I'm now using this approach. @AlekseyTs Let me know what do you think. Thanks!

Reply:
I see there are legitimate test failures. I'll fix them.